### DISCUSSION TITLE
OpenGL ES on windows

### DISCUSSION THREAD

**Discussion by vpenades** [dotnet/Silk.NET#1309] [Q&A]
I always understood that OpenGL was for desktop and OpenGL ES was for mobile.

But given that OpenGL ES is a subset of OpenGL, I guess it could be possible to run on desktop?

In other words, my question is:  Does Silk.Net.OpenGLES  work on desktop (as long as mobile) ? or it is expected-recomended-onlyAvailable on mobile?

**Comment by Perksey** [comment]
This depends on your driver but yes in theory.

**Comment by vpenades** [comment]
My question, and what I wanted to discuss is, given that all the flavours of OpenGL are partially cross platform, but the APIs overlap quite a lot between all these flavours... if there's one API or a way to write code for one api that fits all, so you only write code for a single API, and it works on desktop, web, and smart devices.

Obviously, such an API surface would be more limited than desktop API, but few applications usually require cutting edge graphics. It's really painful to write cross platform graphics when you only want to render some scientific graphics.

This issue is partially related to having OpenGL in a WFP/WinUI/MAUI control... ideally, it could be nice to have a single cross platform MAUI control with a single OpenGL API that could be usable across all supported platforms

**Reply by Beyley** [reply]
best bet for going super cross platform with one codebase is targetting GLES3.0 using ANGLE on desktop and native GLES everywhere else

not all desktop GPUs support GLES natively

**Reply by vpenades** [reply]
I understand, in order to do so, I would need to reference `Silk.NET.OpenGLES.ANGLE.Native` and `Silk.NET.OpenGLES` , but I suspect it's more complicated than that?

**Reply by Perksey** [reply]
To be fair, you can technically just use Silk.NET.OpenGLES with a desktop OpenGL context. There are only a few divergent functions. This is generally my tactic to write-once-run-everywhere. We're consolidating all the packages in 3.0 so this shouldn't be an issue in the distant future.

**Reply by HurricanKai** [reply]
We have talked about making the API objects for GL (but also vulkan, DX, etc.) implement interfaces for feature groups. But this seems fairly unlikely to happen for 2.x and won't be in 3.0 either (maybe eventually 3.x?)
